Content.Server/Pinpointer/PinpointerSystem.cs

```diff
@@ -187,12 +187,11 @@ protected override void UpdateDirectionToTarget(EntityUid uid, PinpointerCompone
 
     private Distance CalculateDistance(Vector2 vec, PinpointerComponent pinpointer)
     {
-        var dist = vec.Length();
-        if (dist <= pinpointer.ReachedDistance)
+        if (vec.IsShorterThanOrEqualTo(pinpointer.ReachedDistance))
             return Distance.Reached;
-        else if (dist <= pinpointer.CloseDistance)
+        else if (vec.IsShorterThanOrEqualTo(pinpointer.CloseDistance))
             return Distance.Close;
-        else if (dist <= pinpointer.MediumDistance)
+        else if (vec.IsShorterThanOrEqualTo(pinpointer.MediumDistance))
```

Comment:
This is slower than what we had before. You overestimate how expensive sqrt is vs `x^2 + y^2`

For a rough approximation, the cost of the sqrt is that of the two squaring operations (not the addition). It's very minor and only adds up in hot loops.